src/middlewared/middlewared/plugins/mdns.py

```diff
@@ -1,14 +1,62 @@
 import threading
 import time
+import enum
 import os
 import pybonjour
 import select
-import socket
 import subprocess
 
 from bsd.threading import set_thread_name
 
 from middlewared.service import Service, private
+from middlewared.utils import filter_list
+
+
+class DevType(enum.Enum):
+    AIRPORT = 'AirPort'
+    APPLETV = 'AppleTv1,1'
+    MACPRO = 'MacPro'
+    RACKMAC = 'RackMac'
+    TIMECAPSULE = 'TimeCapsule6,106'
+    XSERVE = 'Xserve'
+
+    def __str__(self):
+        return self.value
+
+
+class ServiceType(enum.Enum):
+    ADISK = ('_adisk._tcp.', 9)
+    AFPOVERTCP = ('_afpovertcp._tcp.', 548)
+    DEV_INFO = ('_device-info._tcp.', 9)
+    FTP = ('_ftp._tcp.', 21)
+    HTTP = ('_http._tcp.', 80)
+    HTTPS = ('_https._tcp.', 443)
+    ISCSI = ('_iscsi._tcp.', 3260)
+    MIDDLEWARE = ('_middleware._tcp.', 6000)
+    MIDDLEWARE_SSL = ('_middleware-ssl._tcp.', 443)
+    NFS = ('_nfs._tcp.', 2049)
+    SSH = ('_ssh._tcp.', 22)
+    SFTP_SSH = ('_sftp-ssh._tcp.', 22)
+    SMB = ('_smb._tcp.', 445)
+
+
+class SrvToThread(enum.Enum):
+    ADISK = 'mDNSServiceAdiskThread'
+    AFP = 'mDNSServiceAFPThread'
+    DEV_INFO = 'mDNSServiceDevInfoThread'
+    FTP = 'mDNSServiceFTPThread'
+    HTTP = 'mDNSServiceHTTPThread'
+    HTTPS = 'mDNSServiceHTTPSThread'
+    ISCSI = 'mDNSServiceISCSIThread'
+    MIDDLEWARE = 'mDNSServiceMiddlewareThread'
+    MIDDLEWARE_SSL = 'mDNSServiceMiddlewareSSLThread'
+    NFS = 'mDNSServiceNFSThread'
+    SSH = 'mDNSServiceSSHThread'
+    SFTP_SSH = 'mDNSServiceSFTP_SSHThread'
+    SMB = 'mDNSServiceSMBThread'
```

Comment:
Is it a requirement to have a thread for each service?
If not, would it be too much work to have them in the same?